lib/ansible/modules/yum_repository.py

```diff
@@ -657,6 +676,69 @@ def main():
     if module.params['params']:
         module.fail_json(msg="The params option to yum_repository was removed in Ansible 2.5 since it circumvents Ansible's option handling")
 
+    if module.params['name'] is None and module.params['copr'] is None:
+        module.fail_json(msg="Either the 'name' or the 'copr' parameter needs to be set")
+
+    if module.params['copr']:
+        copr_api = module.params['copr_api']
+        try:
+            copr_user, copr_project = module.params['copr'].split('/')
+        except ValueError:
+            module.fail_json(msg="failed to parse {0}".format(module.params['copr']))
+        copr_server = generic_urlparse(list(urlparse(copr_api)))['hostname']
+        distro = get_distribution().lower()
+        if distro in ['centos', 'rhel', 'redhat']:
```

Comment:
I talked with some of the team about this -- eventually I'm going to pull out the `OS_FAMILY_MAP` from `module_utils/facts/system/distribution.py` into something that we can use in situations like this (otherwise -- if we hardcode everywhere like this, we miss things like scientific linux and amazon linux which are also "RHEL-ish").

So this is fine for now, I want to get this merged, but can we add a TODO to abstract this more later?

```suggestion
        # TODO: Abstract OS_FAMILY_MAP from distribution facts and use the appropriate family instead.
        if distro in ['centos', 'rhel', 'redhat']:
```